lib/mixins/vector.js

```diff
@@ -63,20 +63,18 @@ export default {
 
   dash(length, options = {}) {
     let phase;
-    if (length == null) {
-      return this;
+    const originalLength = length;
+    if (!Array.isArray(length)) {
+      length = [length, options.space || length];
     }
-    if (Array.isArray(length)) {
-      length = length.map(v => number(v)).join(' ');
-      phase = options.phase || 0;
-      return this.addContent(`[${length}] ${number(phase)} d`);
-    } else {
-      const space = options.space != null ? options.space : length;
-      phase = options.phase || 0;
-      return this.addContent(
-        `[${number(length)} ${number(space)}] ${number(phase)} d`
-      );
+
+    const valid = length.every(x => Number.isFinite(x) && x > 0);
+    if(!valid) {
+      throw new Error(`dash(${JSON.stringify(originalLength)}, ${JSON.stringify(options)}) invalid, lengths must be numeric and greater than zero`);
     }
+    length = length.map(v => number(v)).join(' ');
+    phase = options.phase || 0;
+    return this.addContent(`[${length}] ${number(phase)} d`);
```

Comment:
```suggestion
    return this.addContent(`[${length}] ${number(options.phase || 0)} d`);
```
And remove phase variable